servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SourceAdapters.java

```diff
@@ -133,39 +135,75 @@ private static <T> SingleSource<T> uncheckedCast(final Single<T> single) {
         return (SingleSource<T>) single;
     }
 
-    private static final class PublisherToPublisherSource<T> implements PublisherSource<T> {
+    // Visible for testing
+    static final class PublisherToPublisherSource<T> extends Publisher<T> implements PublisherSource<T> {
         private final Publisher<T> publisher;
 
         PublisherToPublisherSource(final Publisher<T> publisher) {
             this.publisher = requireNonNull(publisher);
         }
 
+        @Override
+        protected void handleSubscribe(final Subscriber<? super T> subscriber) {
+            deliverErrorFromSource(subscriber, newUnsupportedOperationException(getClass()));
+        }
+
+        @Override
+        void handleSubscribe(final Subscriber<? super T> subscriber,
+                             final CapturedContext capturedContext, final AsyncContextProvider contextProvider) {
+            publisher.handleSubscribe(subscriber, capturedContext, contextProvider);
+        }
+
         @Override
         public void subscribe(final Subscriber<? super T> subscriber) {
             publisher.subscribeInternal(subscriber);
         }
     }
 
-    private static final class SingleToSingleSource<T> implements SingleSource<T> {
+    // Visible for testing
+    static final class SingleToSingleSource<T> extends Single<T> implements SingleSource<T> {
         private final Single<T> single;
 
         SingleToSingleSource(final Single<T> single) {
             this.single = requireNonNull(single);
         }
 
+        @Override
+        protected void handleSubscribe(final Subscriber<? super T> subscriber) {
+            deliverErrorFromSource(subscriber, newUnsupportedOperationException(getClass()));
+        }
+
+        @Override
+        void handleSubscribe(final Subscriber<? super T> subscriber,
+                             final CapturedContext capturedContext, final AsyncContextProvider contextProvider) {
+            single.handleSubscribe(subscriber, capturedContext, contextProvider);
```

Comment:
It shouldn't be part of this PR, but can we delete `delegateSubscribe(a, b, c)` in a followup? It is package private and final, only to forwarding calls to `handleSubscribe(a, b, c)` and half the time we just go straight  to the latter.

Reply:
Good point. Actually, I didn't pay much attention to `delegateSubscribe` vs `handleSubscribe`. 
In this PR I made the opposite change and used `delegateSubscribe` instead to make it consistent with all the other places.

Somehow, I like an idea of having a clear method name that operators can call to delegate subscribe to another source, while `handleSubscribe(a, b, c)` is used only for overrides. Even though it looks like a redundant call in the stack-trace. I'm open to reconsider in the future if we decide to kill `delegateSubscribe`.

Reply:
We can come back to it.